commit 76ef9e1b638268bbe71d0ae75b30880fc48a3a93
Author: Radek Novacek <rnovacek@redhat.com>
Date:   Thu Apr 2 12:41:50 2015 +0200

    ESX: relogin after connection failure
    
    When we get an error from ESX connection, we should always relogin,
    the error might mean that vCenter/ESXi server was restarted and login session is lost.

diff --git a/virt/esx/esx.py b/virt/esx/esx.py
index 63778d5..bd30803 100644
--- a/virt/esx/esx.py
+++ b/virt/esx/esx.py
@@ -26,6 +26,7 @@ from datetime import datetime
 from urllib2 import URLError
 import socket
 from collections import defaultdict
+from httplib import HTTPException
 
 import virt
 
@@ -47,13 +48,17 @@ class Esx(virt.Virt):
 
         self.filter = None
 
-    def _run(self):
+    def _prepare(self):
+        """ Prepare for obtaining information from ESX server. """
         self.logger.debug("Log into ESX")
         self.login()
 
         self.logger.debug("Creating ESX event filter")
         self.filter = self.createFilter()
 
+    def _run(self):
+        self._prepare()
+
         version = ''
         self.hosts = defaultdict(Host)
         self.vms = defaultdict(VM)
@@ -90,13 +95,15 @@ class Esx(virt.Virt):
                 # Get the initial update again
                 version = ''
                 continue
-            except suds.WebFault:
+            except (suds.WebFault, HTTPException):
                 self.logger.exception("Waiting for ESX events fails:")
                 try:
                     self.client.service.CancelWaitForUpdates(_this=self.sc.propertyCollector)
                 except Exception:
                     pass
                 version = ''
+                self._prepare()
+                start_time = end_time = datetime.now()
                 continue
 
             if updateSet is not None:
